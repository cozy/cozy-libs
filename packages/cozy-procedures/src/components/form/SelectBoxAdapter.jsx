import React from 'react'
import PropTypes from 'prop-types'
import { SelectBox, translate } from 'cozy-ui/transpiled/react'

import getInputProps from './getInputProps'

export const SelectBoxAdapter = ({
  onChange,
  options,
  t,
  value,
  ...otherProps
}) => {
  const currentValue = options.enumOptions.find(o => o.value === value)
  let translatedCurrentValue
  if (currentValue) {
    translatedCurrentValue = {
      ...currentValue,
      label: t(currentValue.label)
    }
  }

  return (
    <SelectBox
      {...getInputProps(otherProps, 'select')}
      fullwidth
      options={options.enumOptions.map(option => ({
        ...option,
        label: t(option.label)
      }))}
      onChange={option => onChange(option.value)}
      value={translatedCurrentValue}
    />
  )
}

const ValuePropType = PropTypes.oneOfType([PropTypes.number, PropTypes.string])

SelectBoxAdapter.propTypes = {
  onChange: PropTypes.func.isRequired,
  options: PropTypes.shape({
    enumOptions: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string,
        value: ValuePropType
      })
    )
  }).isRequired,
  t: PropTypes.func.isRequired,
  value: ValuePropType
}

export default translate()(SelectBoxAdapter)
