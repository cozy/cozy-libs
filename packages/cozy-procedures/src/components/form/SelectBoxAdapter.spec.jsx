import React from 'react'

import tMock from '../../../test/__mocks__/tMock'
import { SelectBoxAdapter } from './SelectBoxAdapter'

describe('SelectBoxAdapter component', () => {
  it('should match snapshot', () => {
    const onChangeSpy = jest.fn()
    const props = {
      onChange: onChangeSpy,
      options: {
        enumOptions: [
          { label: 'Foo', value: 'foo' },
          { label: 'Bar', value: 'bar' }
        ],
        isClearable: true,
        isSearchable: false
      },
      t: tMock,
      value: 'foo',
      schema: {
        type: 'string'
      }
    }
    const component = shallow(<SelectBoxAdapter {...props} />)
    expect(component).toMatchSnapshot()
  })
})
