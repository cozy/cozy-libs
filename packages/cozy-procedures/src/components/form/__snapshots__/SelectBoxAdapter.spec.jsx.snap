// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBoxAdapter component should match snapshot 1`] = `
<Aware
  fullwidth={true}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Foo ",
        "value": "foo",
      },
      Object {
        "label": "Bar ",
        "value": "bar",
      },
    ]
  }
  value={
    Object {
      "label": "Foo ",
      "value": "foo",
    }
  }
/>
`;
